### ISSUE TITLE
Outline the Various Applications of IPFS, with Use Cases and Dependencies

### ISSUE THREAD

**Issue by flyingzumwalt** [ipfs/ipfs#230]
For each of the [applications of ipfs that we've made issues for](https://github.com/ipfs/ipfs/labels/applications%20of%20ipfs) provide an outline of:
* use cases (very granular)
* groupings of use cases (coarser, clearer)
* which foundational features & functionality the use cases need or rely on (ie. libp2p, private networks, etc.) - _this sources interdependence between feature sets_
* related products or projects that are addressing those use cases (ie. matrix working on p2p chat)
* related organizations who either 1) have those use cases or 2) are building solutions for those use cases (ie. akasha, mediachain, openbazaar, neocities)

Guidelines:
* Emphasize specific, deliverable user experiences - not just merged, not just a demo. Actual user experiences.
* If a use case relies on an underlying library, link to the repo on github
* If a use case relies on features or functionality that haven't been implemented or released yet, make sure there is a github issue in the appropriate repo for that feature and link to that issue from the use case.

**Comment by empaempa** [comment]
Would be interesting to outline how a YouTube would work - especially statistics such as number of views, geo data and other relevant data for publishers. 

(In my mind, you should be able to put out any piece of content and be able to watch how it's being consumed)

**Comment by DevonJames** [comment]
Here's a little demo of one implementation of a "decentralized YouTube" that uses IPFS for media storage/distrib (although we are not yet capturing views, geo data et cetera)
https://alexandria.io/browser/002004

**Comment by empaempa** [comment]
Nice! Is it possible to capture views/downloads "server side" so to speak (obviously possibly in the playback client). I haven't found anything in the API about this...

**Comment by Beanow** [comment]
While similar to #246 Package Management. Software distribution in general is a prime candidate for IPFS I think. An IPFS based 3rd party Android store would be excellent. The main required properties would be simple to do.

**Developers**
- Can publish APK files directly to IPFS as-is for the actual data.
- Can use IPNS to update release info, app description, screenshots, etc. Which conveniently provides a form of code signing by extension.

**App store service**
- Can provide a web app and native app using IPFS to navigate search indexes and app descriptions.
- Can provide additional security by co-signing keys/releases (for example after a review process and/or automated malware scan), without having to duplicate any underlying data.
- Can provide recovery procedures should developers lose the key to publish releases under.
- Can seed relevant data and provide gateways.

**Comment by DevonJames** [comment]
Yes, that's the plan, and then supplement that with other non server-focused stats that can help gauge the relative popularity of a piece of content like size of swarm and total revenue collected for it to date

**Comment by whyrusleeping** [comment]
These are great conversations to have, Can we try to move them to a more specifically relevant issue? This thread will get quite large and hard to follow otherwise

**Comment by deadlyicon** [comment]
@DevonJames got an updated link?

**Comment by jessicaschilling** [comment]
Note: Discussion on applications of IPFS are happening over in the [IPFS Forums](https://discuss.ipfs.io/c/ecosystem/applications-of-ipfs/) now ... please continue the discussion there! 

This issue is being moved over to the archived repo https://github.com/ipfs/apps/ for reference.
